src/configuration.ts

```diff
@@ -1243,23 +1246,13 @@ export async function initFromSettings(activation: boolean = false): Promise<voi
             }
 
             subKey = "configurationCachePath";
-            let updatedConfigurationCachePath : string | undefined = await util.getExpandedSetting<string>(subKey);
-            if (updatedConfigurationCachePath) {
-                // If there is a directory defined within the configuration cache path,
-                // honor it and don't append to extensionOutputFolder.
-                let parsePath: path.ParsedPath = path.parse(updatedConfigurationCachePath);
-                if (extensionOutputFolder && !parsePath.dir) {
-                    updatedConfigurationCachePath = path.join(extensionOutputFolder, updatedConfigurationCachePath);
-                } else {
-                    updatedConfigurationCachePath = util.resolvePathToRoot(updatedConfigurationCachePath);
-                }
-            }
-            if (updatedConfigurationCachePath !== configurationCachePath) {
```

Comment:
Is there some duplicated code here? It seems like we do similar checks in both this section and the `readConfigurationCachePath` method as well, could they be shared or combined?

Reply:
There was duplication before these diffs, the red part. I'm not sure I see duplication now. Similar changes might be needed for other fragments up and down but I did not focus on other settings to cleanup. Let me know if this is what you meant.